Sources/Commands/SwiftTool.swift

```diff
@@ -44,9 +47,17 @@ private class ToolWorkspaceDelegate: WorkspaceDelegate {
         let totalBytesToDownload: Int64
     }
 
+    private struct FetchProgress {
+        let objectsFetched: Int
+        let totalObjectsToFetch: Int
+    }
+
     /// The progress of each individual downloads.
     private var downloadProgress: [String: DownloadProgress] = [:]
 
+    /// The progress of each individual fetch operation
+    private var fetchProgress: [String: FetchProgress] = [:]
```

Comment:
does this needs to be thread safe? if so use `ThreadSafeKeyValueStore`

Reply:
`fetchProgress` is only accessed from within the "org.swift.swiftpm.commands.tool-workspace-delegate" queue, this shouldn't need to be thread safe.

Reply:
Indeed, it seems to be exactly along the same lines as `downloadProgress` above.